parlai/core/metrics.py

```diff
@@ -307,22 +326,26 @@ def update(self, observation, labels):
                 self.metrics['correct_cnt'] += 1
 
             # F1 and BLEU metrics.
-            f1 = _f1_score(prediction, labels)
-            bleu = _bleu(prediction, labels)
-            rouge1, rouge2, rougel = _rouge(prediction, labels)
+            if 'f1' in self.metrics_list:
+                f1 = _f1_score(prediction, labels)
+            if 'bleu' in self.metrics_list:
+                bleu = _bleu(prediction, labels)
+            if 'rouge-L' in self.metrics_list:
```

Comment:
given that you only check `if 'rouge-L' in metrics_list`, i'd prefer if we just had one key for all of the rouges. `if 'rouge' in metrics_list:`. Maybe @emilydinan  or @jaseweston will find that too obtuse though.

Reply:
yes i'd prefer that

Reply:
I am not sure that is a good idea, you will need to deal with every rouge inside it separately, instead of simplying looping inside self.metrics_list when you init the self.metrics dict....

Reply:
you enumerate them all anyway... Let's do it the single key way, please.

Reply:
Sure.